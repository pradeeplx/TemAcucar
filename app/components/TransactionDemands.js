import React, { Component } from 'react'
import { View, ActivityIndicator } from 'react-native'
import LoadMore from "../components/LoadMore"
import TransactionDemand from "../components/TransactionDemand"
import NoTransactionDemands from "../components/NoTransactionDemands"

export default TransactionDemands = ({ onList, onView, onViewDemand, onComplete, onCancel, onReactivate, demands, listing, canList, currentUser }) => (
  <View >
    { demands.map((demand, index) => (
      <TransactionDemand
        key={demand.id}
        demand={demand}
        index={index}
        currentUser={currentUser}
        onView={onView}
        onViewDemand={onViewDemand}
        onComplete={onComplete}
        onCancel={onCancel}
      />
    )) }
    { listing && <ActivityIndicator style={{ marginTop: 10 }} /> }
    { demands.length === 0 && !listing && <NoTransactionDemands /> }
    { canList && !listing &&
      <LoadMore onPress={onList} />
    }
  </View>
)
