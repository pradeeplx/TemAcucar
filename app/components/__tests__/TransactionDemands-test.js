import 'react-native'
import React from 'react'
import TransactionDemands from '../TransactionDemands'
import renderer from 'react-test-renderer'

const props = {
  style: {top: 0},
  demands: [{
    name: "foo",
    id: 1,
    user: {
      id: 1,
    },
    transactions: [{
      id: 1, user: { id: 1 }, demand: { user: { id: 2 } }
    }, {
      id: 2, user: { id: 1 }, demand: { user: { id: 1 } }
    }]
  }],
  currentUser: { id: 1 },
}

it('renders correctly', () => {
  const tree = renderer.create(
    <TransactionDemands {...props} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders ActivityIndicator', () => {
  const tree = renderer.create(
    <TransactionDemands {...props} demands={[]} listing />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders NoTransactionDemands', () => {
  const tree = renderer.create(
    <TransactionDemands {...props} demands={[]} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders LoadMore button', () => {
  const tree = renderer.create(
    <TransactionDemands {...props} demands={[]} canList />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
