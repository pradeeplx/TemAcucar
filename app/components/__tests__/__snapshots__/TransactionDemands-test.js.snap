exports[`test renders ActivityIndicator 1`] = `
<View>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
    style={
      Object {
        "marginTop": 10,
      }
    } />
</View>
`;

exports[`test renders LoadMore button 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#654C50",
          "fontFamily": "Avenir",
          "fontSize": 18,
        },
        Object {
          "alignSelf": "stretch",
          "fontSize": 15,
          "margin": 10,
          "textAlign": "center",
        },
      ]
    }>
    Você ainda não realizou nenhuma transação. Que tal começar agora mesmo fazendo um pedido?
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Array [
            Object {
              "color": "#654C50",
            },
            Array [
              Object {
                "marginBottom": 20,
                "marginTop": 10,
                "textAlign": "center",
              },
              undefined,
            ],
          ],
        ]
      }>
      Carregar mais
    </Text>
  </View>
</View>
`;

exports[`test renders NoTransactionDemands 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#654C50",
          "fontFamily": "Avenir",
          "fontSize": 18,
        },
        Object {
          "alignSelf": "stretch",
          "fontSize": 15,
          "margin": 10,
          "textAlign": "center",
        },
      ]
    }>
    Você ainda não realizou nenhuma transação. Que tal começar agora mesmo fazendo um pedido?
  </Text>
</View>
`;

exports[`test renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ECF8F7",
          "opacity": 1,
          "padding": 20,
        }
      }
      testID={undefined}>
      <Image
        size={48}
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 36,
              "height": 72,
              "width": 72,
            },
            undefined,
          ]
        } />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Avenir",
              "fontSize": 18,
            },
            Array [
              Object {
                "color": "#44B5B1",
                "fontSize": 18,
                "marginTop": 10,
              },
              Object {
                "fontFamily": "Avenir-Black",
              },
            ],
          ]
        }>
        foo
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#654C50",
                "fontFamily": "Avenir",
                "fontSize": 18,
              },
              Object {
                "color": "#000000",
                "fontSize": 15,
              },
            ]
          }>
          Você
           
          pediu 
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#654C50",
                "fontFamily": "Avenir",
                "fontSize": 18,
              },
              Array [
                Object {
                  "fontSize": 13.5,
                },
                Object {
                  "color": "#000000",
                  "fontSize": 15,
                },
              ],
            ]
          }>
          há segundos
        </Text>
      </View>
    </View>
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#EFEDED",
            "borderTopWidth": 0,
            "flexDirection": "row",
            "opacity": 1,
            "padding": 10,
          }
        }
        testID={undefined}>
        <Image
          size={32}
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Array [
              Object {
                "borderRadius": 24,
                "height": 48,
                "width": 48,
              },
              Object {
                "marginRight": 10,
              },
            ]
          } />
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#654C50",
                  "fontFamily": "Avenir",
                  "fontSize": 18,
                },
                Object {
                  "color": "#654C50",
                },
              ]
            }>
            undefined undefined
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#654C50",
                  "fontFamily": "Avenir",
                  "fontSize": 18,
                },
                Object {
                  "color": "#CECED2",
                },
              ]
            }>
            Escreva uma mensagem para undefined
          </Text>
        </View>
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#654C50",
                "fontFamily": "Ionicons",
                "fontSize": 27,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "color": "#CECED2",
              },
            ]
          }>
          
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#EFEDED",
            "borderTopWidth": 1,
            "flexDirection": "row",
            "opacity": 1,
            "padding": 10,
          }
        }
        testID={undefined}>
        <Image
          size={32}
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Array [
              Object {
                "borderRadius": 24,
                "height": 48,
                "width": 48,
              },
              Object {
                "marginRight": 10,
              },
            ]
          } />
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#654C50",
                  "fontFamily": "Avenir",
                  "fontSize": 18,
                },
                Object {
                  "color": "#654C50",
                },
              ]
            }>
            undefined undefined
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#654C50",
                  "fontFamily": "Avenir",
                  "fontSize": 18,
                },
                Object {
                  "color": "#CECED2",
                },
              ]
            }>
            Escreva uma mensagem para undefined
          </Text>
        </View>
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#654C50",
                "fontFamily": "Ionicons",
                "fontSize": 27,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "color": "#CECED2",
              },
            ]
          }>
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
